source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -388,6 +389,29 @@ void ClientSslSocketFactory::addReadyCb(std::function<void()> callback) {
   }
 }
 
+bool ClientSslSocketFactory::secureTransportReady() const {
+  if (!implementsSecureTransport()) {
+    return false;
+  }
+  // If there is no secret entity, currently supports only TLS Certificate and Validation
+  // Context, when it failed to extract them via SDS, it will fail to change cluster status from
+  // warming to active. In current implementation, there is no strategy to activate clusters
+  // which failed to initialize at once.
+  // TODO(shikugawa): Consider retry strategy of clusters which failed to activate at once.
+  const auto& tls_certificate_sds_configs = config_->tlsCertificateSdsConfigs();
+  for (const auto& config : tls_certificate_sds_configs) {
+    if (!secret_manager_.checkTlsCertificateEntityExists(config.sds_config(), config.name())) {
```

Comment:
I think we can use `tls_certificate_providers_` from ContextConfigImpl (you may need to expose it via methods) to do this?

Reply:
done